### PR TITLE
kubeadm: increase ut coverage for bootstraptoken/node

### PR DESCRIPTION
**Pull request by my-git9**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Increase ut coverage for bootstraptoken/node

before:
```
        k8s.io/kubernetes/cmd/kubeadm/app/phases/bootstraptoken/node    coverage: 45.8% of statements
```

after this change:
```
        k8s.io/kubernetes/cmd/kubeadm/app/phases/bootstraptoken/node    coverage: 83.3% of statements
```

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go [hunk 1]
```diff
@@ -0,0 +1,305 @@
+/*
+Copyright 2017 The Kubernetes Authors.
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go:2]:
```suggestion
Copyright 2023 The Kubernetes Authors.
```

File: cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+)
+
+func TestAllowBootstrapTokensToPostCSRs(t *testing.T) {
+	tests := []struct {
+		name    string
+		client  clientset.Interface
+		wantErr bool
+	}{
+		{
+			name:   "ClusterRoleBindding is empty",
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go:37]: please change `s/Bidding/Binding` in the file.

**Review by neolit123 — COMMENTED** [review]
/approve

File: cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+	clientset "k8s.io/client-go/kubernetes"
+	clientsetfake "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/kubernetes/cmd/kubeadm/app/constants"
+)
+
+func TestAllowBootstrapTokensToPostCSRs(t *testing.T) {
+	tests := []struct {
+		name    string
+		client  clientset.Interface
+		wantErr bool
```
Comment (chendave) [inline cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go:34]:
```suggestion

```

File: cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+	}{
+		{
+			name:   "ClusterRoleBinddings are empty",
```
Comment (chendave) [inline cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go:37]:
```suggestion
			name:   "ClusterRoleBinding is empty",
```

File: cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+			client: clientsetfake.NewSimpleClientset(),
+		},
+		{
+			name: "ClusterRoleBinddings are consistent with initialization",
```
Comment (chendave) [inline cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go:41]:
```suggestion
			name: "ClusterRoleBindings is already exists",
```

File: cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+				Subjects: []rbac.Subject{
+					{
+						Kind: rbac.GroupKind,
+						Name: constants.NodeBootstrapTokenAuthGroup,
+					},
+				},
+			}),
+		},
+		{
+			name: "ClusterRoleBinddings are different from initialization",
```
Comment (chendave) [inline cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go:60]:
```suggestion
			name: "Create new ClusterRoleBinddings",
```

File: cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+						Name: constants.KubeProxyClusterRoleName,
+					},
+				},
+			}),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := AllowBootstrapTokensToPostCSRs(tt.client); err != nil {
+				t.Errorf("AllowBootstrapTokensToPostCSRs() error = %v", err)
```
Comment (chendave) [inline cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go:82]:
```suggestion
				t.Errorf("AllowBootstrapTokensToPostCSRs() return error = %v", err)
```

File: cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go [hunk 8]
```diff
@@ -0,0 +1,305 @@
...
+			}
+		})
+	}
+}
+
+func TestAutoApproveNodeBootstrapTokens(t *testing.T) {
+	tests := []struct {
+		name    string
+		client  clientset.Interface
+		wantErr bool
```
Comment (chendave) [inline cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go:92]:
```suggestion
```

Comment (chendave) [reply cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go:92]: and update this for all other testcase where this field is not used at all.

File: cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go [hunk 9]
```diff
@@ -0,0 +1,305 @@
...
+	}{
+		{
+			name:   "ClusterRoleBinddings are empty",
+			client: clientsetfake.NewSimpleClientset(),
+		},
+		{
+			name: "ClusterRoleBinddings are consistent with initialization",
```
Comment (chendave) [inline cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go:99]: pls rename all those testcases.

File: cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go [hunk 10]
```diff
@@ -0,0 +1,305 @@
...
+						Name: constants.NodeBootstrapTokenAuthGroup,
+					},
+				},
+			}),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := AutoApproveNodeBootstrapTokens(tt.client); err != nil {
+				t.Errorf("AutoApproveNodeBootstrapTokens() error = %v", err)
```
Comment (chendave) [inline cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go:140]: ... return error, and pls fix all other occurrences.

File: cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go [hunk 11]
```diff
@@ -0,0 +1,301 @@
...
+	"k8s.io/kubernetes/cmd/kubeadm/app/constants"
+)
+
+func TestAllowBootstrapTokensToPostCSRs(t *testing.T) {
+	tests := []struct {
+		name   string
+		client clientset.Interface
+	}{
+		{
+			name:   "ClusterRoleBinddings is empty",
```
Comment (chendave) [inline cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go:36]: pls replace all the ClusterRoleBinddings with ClusterRoleBindings

File: cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go [hunk 12]
```diff
@@ -0,0 +1,301 @@
...
+						Name: constants.NodeBootstrapTokenAuthGroup,
+					},
+				},
+			}),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := AutoApproveNodeBootstrapTokens(tt.client); err != nil {
+				t.Errorf("AutoApproveNodeBootstrapTokens() error = %v", err)
```
Comment (chendave) [inline cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go:138]: Can you update this as well?

File: cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go [hunk 13]
```diff
@@ -0,0 +1,301 @@
...
+	}
+}
+
+func TestAllowBoostrapTokensToGetNodes(t *testing.T) {
+	tests := []struct {
+		name   string
+		client clientset.Interface
+	}{
+		{
+			name:   "Rbac rules are empty",
```
Comment (chendave) [inline cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go:207]: pls upper case as "RBAC"

File: cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go [hunk 14]
```diff
@@ -0,0 +1,301 @@
...
+	tests := []struct {
+		name   string
+		client clientset.Interface
+	}{
+		{
+			name:   "ClusterRoleBindings is empty",
+			client: clientsetfake.NewSimpleClientset(),
+		},
+		{
+			name: "ClusterRoleBindings is already exists",
```
Comment (chendave) [inline cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go:97]:
```suggestion
			name: "ClusterRoleBindings already exists",
```

File: cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go [hunk 15]
```diff
@@ -0,0 +1,301 @@
...
```
Comment (chendave) [inline cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go:40]:
```suggestion
			name: "ClusterRoleBindings already exists",
```

File: cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go [hunk 16]
```diff
@@ -0,0 +1,301 @@
...
```
Comment (chendave) [inline cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go:154]:
```suggestion
			name: "ClusterRoleBindings already exists",
```

File: cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go [hunk 17]
```diff
@@ -0,0 +1,301 @@
...
+	tests := []struct {
+		name   string
+		client clientset.Interface
+	}{
+		{
+			name:   "Rbac rules are empty",
+			client: clientsetfake.NewSimpleClientset(),
+		},
+		{
+			name: "Rbac rules are already exists",
```
Comment (chendave) [inline cmd/kubeadm/app/phases/bootstraptoken/node/tlsbootstrap_test.go:211]:
```suggestion
			name: "Rbac rules already exist",
```

**Review by chendave — COMMENTED** [review]
Just some syntax suggestion!

/lgtm
/hold

you can lift the hold as your preference.

thanks, let's keep going!

**Review by chendave — COMMENTED** [review]
/lgtm

**Pull request merged**